### ISSUE TITLE
Brush onDragStop

### ISSUE THREAD

**Issue by sharabiania** [recharts/recharts#2534]
Currently Brush only has onChange event, which gets fired numerous times when the user is interacting with the Brush.

How can we fire an even only when the user stops dragging mouse on the brush?

basically an onDragStop() event.

**Comment by bsplosion** [comment]
@xile611 this has been a request for a few years now, see https://github.com/recharts/recharts/issues/859.  Are there any plans for implementing additional brush events?

**Comment by William-limpide** [comment]
We also wanted an event like onDragEnd(), but because it does not exist we had to implement debouncing in the useEffect with setTimeout to keep the brush component from calling the api whenever the value change.

You can adjust the delay value in setTimeout to have what you want.

```
useEffect(() => {
      const getData = setTimeout( ( ) => {
          api.get()
}, delay )

return ( ) => clearTimeout(getData)
}, [startDate, endDate])
```
This way, whenever the user will use the brush, the timeout will get cleared and only the last call will be fired thanks to the delay.

**Comment by simkesd** [comment]
This can now be closed. Check https://github.com/recharts/recharts/pull/3774 for added onDragEnd.

@nicholasgcoles Just a ping so we don't keep this issue opened anymore.

**Comment by ckifer** [comment]
released in 2.9
